.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,25 @@
+name: golangci-lint
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5
+        with:
+          persist-credentials: false
```

Comment:
Hat tip to zizmor for catching the need to put `persist-credentials: false` here, which I would have missed otherwise